webapp/src/pages/UserPage/UserPage.tsx

```diff
@@ -18,28 +18,23 @@ import {
   AVATAR_EP_URL,
 } from '../../shared/urls';
 import { useData } from '../../shared/hooks/UseData';
-import { goBack } from '../../shared/callbacks';
 import { useCallback } from 'react';
-import { useNavigate, useParams } from 'react-router-dom';
+import { useParams } from 'react-router-dom';
 import { useAuth } from '../../shared/hooks/UseAuth';
 import { usersApi } from '../../shared/services/ApiService';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import { User } from '../../shared/generated';
 
-type UserPageProps = {
-  isMe: boolean;
+type UserComponentTemplateProps = {
+  user: User | null;
+  isAuthUser: boolean;
 };
 
-export default function UserPage({ isMe = false }: UserPageProps) {
-  const param = useParams();
-  const getCurrentUser = useCallback(
-    () => usersApi.userControllerGetCurrentUser(),
-    [],
-  );
-  const getUserById = useCallback(
-    () => usersApi.userControllerGetUserById({ userId: param.id! }),
-    [param.id],
-  );
-  const { data: user } = useData(isMe ? getCurrentUser : getUserById);
-  const navigate = useNavigate();
+const UserComponentTemplate = ({
+  user,
+  isAuthUser,
```

Comment:
As I wrote in the issue, where I think it may go unnoticed:

> We have inconsistencies in auth user page: if navigated from a user row, it displays it like a normal user. But if navigate it from user profile, it displays it like a isMe user.

This is intentional. If it's displayed from inside IsMe , it displays all available information. Otherwise, it displays info as it would be available to others. This exists, for instance, in the 42 intra as well.

I honestly don't think it's a good idea to remove the feature of being able to see the user page of the user from both points of view (self, and from the point of view of others)

Reply:
I agree with this :+1:
I think it's nice to have both views.

Reply:
Ok, agree and I've restore the behaviour! I though it was a "side feature" we had due to how our routing was designed rather than a former feature :smile: